app/models/service_template_orchestration.rb

```diff
@@ -24,4 +39,14 @@ def self.default_reconfiguration_entry_point
   def my_zone
     orchestration_manager.try(:my_zone) || MiqServer.my_zone
   end
+
+  def self.validate_config_info(options)
+    config_info = options[:config_info]
+    if (config_info[:template_id] && !config_info[:manager_id]) ||
```

Comment:
You allow a case that neither exists. Let's make it simpler by requiring both must exist.